src/modules/visits/services/visits.service.ts

```diff
@@ -283,21 +289,40 @@ export class VisitsService {
 			}
 
 			// IF accessFolderIds is not empty and accessType is FOLDERS
-			// THEN add these folders to the folder_access table
+			// THEN
+			// - remove existing access folders
+			// - add new folders to the folder_access table
 			if (!isEmpty(accessFolderIds) && updateVisit.access_type === VisitAccessType.Folders) {
-				for (let index = 0; index < accessFolderIds.length; index++) {
-					await this.dataService.execute<
-						InsertVisitFolderAccessMutation,
-						InsertVisitFolderAccessMutationVariables
-					>(InsertVisitFolderAccessDocument, {
-						objects: [
-							...accessFolderIds.map((accessFolderId: string) => ({
-								folder_id: accessFolderId,
-								visit_request_id: currentVisit.id,
-							})),
-						],
-					});
-				}
+				await this.dataService.execute<
+					DeleteVisitFolderAccessMutation,
+					DeleteVisitFolderAccessMutationVariables
+				>(DeleteVisitFolderAccessDocument, {
+					visitRequestId: currentVisit.id,
+				});
+
+				await this.dataService.execute<
+					InsertVisitFolderAccessMutation,
+					InsertVisitFolderAccessMutationVariables
+				>(InsertVisitFolderAccessDocument, {
+					objects: [
+						...accessFolderIds.map((accessFolderId: string) => ({
+							folder_id: accessFolderId,
+							visit_request_id: currentVisit.id,
+						})),
+					],
+				});
+			}
+
+			// IF accessType is FULL
+			// THEN
+			// - remove existing access folders
+			if (updateVisit.access_type === VisitAccessType.Full) {
+				await this.dataService.execute<
+					DeleteVisitFolderAccessMutation,
+					DeleteVisitFolderAccessMutationVariables
+				>(DeleteVisitFolderAccessDocument, {
+					visitRequestId: currentVisit.id,
+				});
```

Comment:
Isn't this shorter if you always remove the folders, and then only add them if access_type is folders?
```
			// Always delete the old access folders:
			await this.dataService.execute<
				DeleteVisitFolderAccessMutation,
				DeleteVisitFolderAccessMutationVariables
			>(DeleteVisitFolderAccessDocument, {
				visitRequestId: currentVisit.id,
			});

			// IF accessFolderIds is not empty and accessType is FOLDERS
			// THEN
			// - add new folders to the folder_access table
			if (!isEmpty(accessFolderIds) && updateVisit.access_type === VisitAccessType.Folders) {
				await this.dataService.execute<
					InsertVisitFolderAccessMutation,
					InsertVisitFolderAccessMutationVariables
				>(InsertVisitFolderAccessDocument, {
					objects: [
						...accessFolderIds.map((accessFolderId: string) => ({
							folder_id: accessFolderId,
							visit_request_id: currentVisit.id,
						})),
					],
				});
			}
```